### PR TITLE
Add pip resolver

### LINKED ISSUE #1642: Workflow linter to analyse PyPI library
Would share significant codebase with:
- https://github.com/databrickslabs/ucx/issues/1640

### PR DESCRIPTION
**Pull request by JCZuurmond**
## Changes
Add PIP resolver

### Linked issues

Resolves #1642

### Functionality 

- [ ] added relevant user documentation
- [ ] added new CLI command
- [ ] modified existing command: `databricks labs ucx ...`
- [ ] added a new workflow
- [ ] modified existing workflow: `...`
- [ ] added a new table
- [ ] modified existing table: `...`

### Tests

- [ ] manually tested
- [ ] added unit tests
- [ ] added integration tests
- [ ] verified on staging environment (screenshot attached)

### CODE REVIEW

File: tests/unit/source_code/test_site_packages.py [hunk 1]
```diff
@@ -1,8 +1,95 @@
...
+
+    assert len(maybe.problems) == 1
+    assert maybe.problems[0] == DependencyProblem("library-install-failed", "Failed to install unknown-library-name")
+
+
 def test_reads_site_packages():
     site_packages_path = locate_site_packages()
     site_packages = SitePackages.parse(str(site_packages_path))
     assert site_packages["astroid"] is not None
+
```
Comment (JCZuurmond) [inline tests/unit/source_code/test_site_packages.py:34]: @ericvergnaud : Below test might become unrelevant given your chances

File: src/databricks/labs/ucx/source_code/site_packages.py [hunk 2]
```diff
@@ -1,14 +1,67 @@
...
+        next_resolver: BaseImportResolver | None = None,
+    ):
+        super().__init__(next_resolver)
+        self._file_loader = file_loader
+
+    def with_next_resolver(self, resolver: BaseImportResolver) -> BaseImportResolver:
+        return PipResolver(self._file_loader, resolver)
+
+    def resolve_import(self, path_lookup: PathLookup, name: str) -> MaybeDependency:
+        path_lookup.append_path(self._temporary_virtual_environment)
```
Comment (JCZuurmond) [inline src/databricks/labs/ucx/source_code/site_packages.py:40]: @ericvergnaud : Should we update the path look outside of each resolver, i.e. on the top level:

```
lib = self._import_resolver.resolve_import()
self._path_look.append_path(lib.path)
return lib
```

**Review by JCZuurmond — COMMENTED** [review]
@ericvergnaud : Please have a look

File: src/databricks/labs/ucx/source_code/site_packages.py [hunk 3]
```diff
@@ -1,14 +1,67 @@
...
     DependencyGraph,
     DependencyProblem,
+    MaybeDependency,
+    SourceContainer,
+    WrappingLoader,
 )
+from databricks.labs.ucx.source_code.path_lookup import PathLookup
+
+
+class PipResolver(BaseImportResolver):
```
Comment (ericvergnaud) [inline src/databricks/labs/ucx/source_code/site_packages.py:22]: I don't think the PipResolver is an ImportResolver. Let's discuss.

**Comment by JCZuurmond** [comment]
Prefer #1694 1694

**Pull request closed without merging**